client/impl/fcm.js

```diff
@@ -32,11 +32,18 @@ function updateToken(firebaseInstance) {
     .then(token => {
       return registerFCMTopic(token);
     })
-    .catch(() => {
+    .catch(err => {
+      console.log(err, "Could not retrieve token from firebase");
       throw new Error("Could not retrieve token from firebase");
```

Comment:
why this log?

Reply:
to find the **Errors inside Promises** when promises are rejected on the client side.

Reply:
We’re throwing an error just after on line 37. Isn’t that enough?

Reply:
got it , have made the changes